controllers/ProposalController.php

```diff
@@ -873,4 +873,142 @@ private function buildNotApprovedProposalsActiveDataProvider(yii\db\ActiveQuery$
             ]
         ]);
     }
+
+    public function actionReviewProposal(int $id)
+    {
+        /** @var \app\models\Proposal $selectedProposal */
+        $selectedProposal = $this->checkIfProposalExists($id);
+        $viewItems = $this->buildOneProposalViewItems($selectedProposal);
+
+        $potentialReview = $this->getPotentialReviewOfAReviewer($selectedProposal, MainController::getCurrentUser()->id);
+        $viewItems['potentialReview'] = $potentialReview;
+
+        return $this->render('proposal', $viewItems);
+    }
+
+    private function getPotentialReviewOfAReviewer(\app\models\Proposal $proposal, $reviewerId)
+    {
+
+        foreach ($proposal->reviews as $review) {
+
+            if ($review->reviewer_id === $reviewerId) {
+                return $review;
+            }
+        }
```

Comment:
I think it is useless to retrieve all reviews of the proposal and then return only one review. You should use `$review = Review::find()->where(...)`.